src/rgw/rgw_compression.cc

```diff
@@ -0,0 +1,165 @@
+                    << "for rgw, check rgw_compression_type config option" << dendl;
+      compressed = false;
+      in_bl.claim(bl);
+    } else {
+      int cr = compressor->compress(bl, in_bl);
+      if (cr < 0) {
+        if (ofs > 0 && compressed) {
+          lderr(cct) << "Compression failed with exit code " << cr
+                     << " for next part, compression process failed" << dendl;
+          return -EIO;
+        }
+        ldout(cct, 5) << "Compression failed with exit code " << cr << dendl;
+        compressed = false;
+        in_bl.claim(bl);
+      } else {
+        compressed = true;
+  
+        compression_block newbl;
+        int bs = blocks.size();
+        newbl.old_ofs = ofs;
+        newbl.new_ofs = bs > 0 ? blocks[bs-1].len + blocks[bs-1].new_ofs : 0;
+        newbl.len = in_bl.length();
+        blocks.push_back(newbl);
+      }
+    }
+  } else {
+    compressed = false;
+    in_bl.claim(bl);
+  }
+  // end of compression stuff
+
+  return next->handle_data(in_bl, ofs, phandle, pobj, again);
+}
+
+//----------------RGWGetObj_Decompress---------------------
+RGWGetObj_Decompress::RGWGetObj_Decompress(CephContext* cct_, 
+                                           const RGWCompressionInfo& cs_info_, 
+                                           bool partial_content_,
+                                           RGWGetDataCB* next): RGWGetObj_Filter(next),
+                                                                cct(cct_),
+                                                                cs_info(cs_info_),
+                                                                partial_content(partial_content_),
+                                                                first_block(0),
+                                                                last_block(0),
+                                                                q_ofs(0),
+                                                                q_len(0),
+                                                                first_data(true),
+                                                                cur_ofs(0)
+{
+  compressor = Compressor::create(cct, cs_info.compression_type);
+  if (!compressor.get())
+    lderr(cct) << "Cannot load compressor of type " << cs_info.compression_type 
+                     << "for rgw, check rgw_compression_type config option" << dendl;
+}
+
+int RGWGetObj_Decompress::handle_data(bufferlist& bl, off_t bl_ofs, off_t bl_len)
+{
+    ldout(cct, 10) << "Compression for rgw is enabled, decompress part " << bl_len << dendl;
+
+    if (!compressor.get()) {
+      // if compressor isn't available - error, because cannot return decompressed data?
+      lderr(cct) << "Cannot load compressor of type " << cs_info.compression_type 
+                       << "for rgw, check rgw_compression_type config option" << dendl;
+      return -EIO;
+    } else {
```

Comment:
don't need an else after the return, this saves some indentation. also, this whole function has two extra spaces of indentation